rancher-sandbox/rancher-desktop
# Add the supervisor service

This PR initializes the[ supervisor privileged service](https://docs.google.com/document/d/1rEMnQJTxJ6oHePrtsdtuES8gi6p_Qibi4BOq1rmzp5E/edit#heading=h.f85ih01fkz16). 

The service listens on a given `host:port` for any published ports for Docker, Containerd from RD Agent (to be implemented).

To test:

- Build a binary 
```
GOOS=windows go build .
``` 
- Locate the binary in a privileged cmd or powershell register the service:
```
privileged-service.exe install
```
Confirm in `services.msc` that the service is installed as `Rancher Desktop Privileged Service`
- Start the service:
```
privileged-service.exe start
```
- Confirm in `services.msc` that the service status is changed to `running`
- Also, confirm the port server is runnig
```
netstat -ano | findstr '4444'
```
- Pause the service
```
privileged-service.exe pause
```
- Confirm that the port server is stopped by running:
```
netstat -ano | findstr '4444'
```
- Continue the service
```
privileged-service.exe continue
```
- Confirm that the port server is running by running:
```
netstat -ano | findstr '4444'
```
- Stop the service 
```
privileged-service.exe stop
```
- Uninstall the service to make sure it is removed from win32 services database.
```
privileged-service.exe uninstall
```
Additionally:

You can test the above star/stops steps from a non-admin user account for the service.


Signed-off-by: Nino Kodabande <nkodabande@suse.com>
